src/drive/ducks/files/Toolbar.jsx

```diff
@@ -131,17 +135,27 @@ class Toolbar extends Component {
             {t('toolbar.menu_select')}
           </a>
         </Item>
-        {shared.withMe && <hr />}
-        {shared.withMe && (
-          <Item>
-            <a
-              className={classNames(styles['fil-action-delete'])}
-              onClick={() => leaveFolder(displayedFolder)}
-            >
-              {t('toolbar.leave')}
-            </a>
-          </Item>
-        )}
+        {notRootfolder && <hr />}
+        {notRootfolder &&
+          (shared.withMe ? (
+            <Item>
+              <a
+                className={classNames(styles['fil-action-delete'])}
+                onClick={() => leaveFolder(displayedFolder)}
+              >
+                {t('toolbar.leave')}
+              </a>
+            </Item>
+          ) : (
+            <Item>
+              <a
+                className={classNames(styles['fil-action-delete'])}
+                onClick={() => leaveFolder(displayedFolder)}
+              >
+                {t('toolbar.trash')}
+              </a>
+            </Item>
+          ))}
```

Comment:
If I'm not missing anything, it's only the label changing here, so we could do `{t(shared.withMe ? 'toolbar.leave' : 'toolbar.trash')}`and leave the rest all the time, no?

Reply:
I forgot the action that changes too.

Reply:
I would like us to sell the whole menu in a more functional way.
We can't have so many conditions and not care that it becomes unreadable and difficult to maintain, when in fact, functionally, it's quite simple: you have buttons that must be present under certain conditions.